.github/workflows/sync.yml

```diff
@@ -0,0 +1,52 @@
+name: Sync
+on:
+  schedule: # every friday at 12:00 UTC
+    - cron: "0 12 * * 5"
+  workflow_dispatch:
+jobs:
+  sync-overview-docs:
+    env:
+      RUSTFLAGS: -A warnings
+      BRANCH: auto/sync-overview-docs
+    permissions:
+      contents: write
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly
+      - uses: Swatinem/rust-cache@v2
+      - run: git checkout -b "${{ env.BRANCH }}"
+      - name: Sync Docs
+        run: cargo xtask overview
+      - name: Configure Git user
+        run: |
+          git config user.name ${{github.actor}}
+          git config user.email "${{github.actor_id}}+${{github.actor}}@users.noreply.github.com"
+      - run: git add .
+      - name: Commit changes
+        run: >
+          git commit -m "docs: sync overview docs"
+      - name: Push changes
+        run: git push origin "${{ env.BRANCH }}" --force
+      - name: Check if PR exists
+        run: gh pr list --state open --base main --head "${{ env.BRANCH }}" --json "id" | jq --exit-status 'length == 0'
+        continue-on-error: true
+        id: pr-list
+        env:
+          GH_TOKEN: ${{ github.token }}
+      # if there is a pr already open, skip the following steps
+      - name: Create PR
+        run: gh pr create --fill --body "Automatic overview sync" --base main --head "${{ env.BRANCH }}"
+        if: steps.pr-list.outcome == 'success'
+        env:
+          GH_TOKEN: ${{ github.token }}
+      - name: Merge PR
+        run: gh pr merge --auto --rebase --delete-branch
```

Comment:
Do we always rebase-merge btw? I'd expect rebasing to not create a merge commit (similar to squash merge).

Reply:
checked, it's set to create a merge commit, not sure how this will work. can try on this pr

Reply:
```
❯ gh pr merge --auto --rebase --delete-branch
! The merge strategy for main is set by the merge queue
✓ Pull request #478 will be added to the merge queue for main when ready
```